tests/unit/test_transport.c

```diff
@@ -0,0 +1,35 @@
+#include "sentry_testsupport.h"
+#include "sentry_transport.h"
+#include "sentry_utils.h"
+#include <sentry.h>
+
+SENTRY_TEST(rate_limit_parsing)
+{
+    uint64_t now = sentry__msec_time();
+    sentry_rate_limiter_t *rl = sentry__rate_limiter_new();
+    TEST_CHECK(sentry__rate_limiter_update_from_header(
+        rl, "120:error:project, 60:session:foo, +30::bar"));
```

Comment:
These tests are outdated. `+30` is no longer supported. Also, could you add some tests for parsing invalid categories and no categories? 

Reference implementation in Relay: https://github.com/getsentry/relay/blob/5069fbc7846782f372bfc0dee522e0aae7d9b673/relay-server/src/utils/rate_limits.rs